LinkedList/Exercise/SLList.h

```diff
@@ -0,0 +1,328 @@
+#pragma once
+
+template<typename Type>
+class TSLList
+{
+private:
+
+	struct FNode
+	{
+		Type Element;
+		FNode* Next = nullptr;
+	};
+
+	FNode* Head;
+	FNode* Tail;
+
+	int ListSize;
+
+	FNode* GetNode(const int Index)
+	{
+		FNode* Current = Head;
+		for (int i = 0; i < Index; i++)
+		{
+			Current = Current->Next;
+		}
+		return Current;
+	}
+
+public:
+
+	class FIterator
+	{
+	private:
+		FNode* Current;
+
+	public:
+
+		FIterator(FNode* NewCurrent)
+		{
+			Current = NewCurrent;
+		}
+
+		FIterator& operator++ ()
+		{
+			Current = Current->Next;
+			return *this;
+		}
+
+		Type& operator* ()
+		{
+			return Current->Element;
+		}
+
+		bool operator!= (const FIterator& Other)
+		{
+			return Current != Other.Current;
+		}
+
+
+	};
+
+	FIterator begin()
+	{
+		if (Head)
+		{
+			return FIterator(Head);
+		}
+		return FIterator(nullptr);
```

Comment:
You can do all of this in a single line. Give it some thought

Reply:
code standards